### PR TITLE
conformance: integer math

### PR DESCRIPTION
**Pull request by madi8229**
Integer math tests for conformance test.

### CODE REVIEW

File: tests/simple/testdata/broken.textproto [hunk 1]
```diff
@@ -61,3 +61,31 @@ section {
     }
   }
 }
+section {
+  name: "integer _math.textproto/Literal Integer Math"
+  description: "tests for integer math."
+  test {
+    name: "int64_overflow"
+    expr: "9223372036854775807 + 1"
```
Comment (JimLarson) [inline tests/simple/testdata/broken.textproto]: Add a "description" field that explains this magic number, such as "INT64_MAX plus one."  Do this for all the tests.

File: tests/simple/testdata/integer_math.textproto [hunk 2]
```diff
@@ -1,2 +1,171 @@
...
+  name: "int64_math"
+  description: "Simple tests for int64."
+  test {
+    name: "simple_add_positive_positive"
+    expr: "42 + 2"
+    value: { int64_value: 44 }
+  }
+  test {
+    name: "simple_add_positive_negative"
+    expr: "42 + (-2)"
```
Comment (JimLarson) [inline tests/simple/testdata/integer_math.textproto]: Change the arguments around a little.  It's still okay to use small numbers.

File: tests/simple/testdata/integer_math.textproto [hunk 3]
```diff
@@ -1,2 +1,171 @@
...
+    expr: "15 / 0"
+    eval_error: {
+      errors: { message:"divide by zero" }
+    }
+  }
+  test {
+    name: "multiply_zero"
+    expr: "15 * 0"
+    value: { int64_value: 0 }
+  }
```
Comment (JimLarson) [inline tests/simple/testdata/integer_math.textproto]: Add the following tests, here and in the following sections as appropriate:
- add_left_identity:  0 + 17 &rarr; 17
- add_right_identity:  83 + 0 &rarr; 83
- add_commutative:  36 + 21 == 21 + 36 &rarr; true
- add_associative: 4 + (15 + 3) == (4 + 15) + 3 &rarr; true (with a note that this is not true when intermediate results might overflow)
- mul_left_identity: 1 * 28 &rarr; 28
- mul_right_identity: -9 * 1 &rarr; -9
- mul_commutative: 39 * 72 == 72 * 39 &rarr; true
- mul_associative: 2 * (7 * -13) == (2 * 7) * -13 &rarr; true (again with a note that this is not true when intermediate results overflow)
- add_mul_distribute: 4 * (2 + 7) == 4 * 2 + 4 *7 &rarr; true

File: tests/simple/testdata/integer_math.textproto [hunk 4]
```diff
@@ -1,2 +1,171 @@
...
+    expr: "42u - 2u"
+    value: { uint64_value: 40 }
+  }
+  test {
+    name: "simple_multiply"
+    expr: "42u * 2u"
+    value: { uint64_value: 84 }
+  }
+  test {
+    name: "simple_devide"
```
Comment (JimLarson) [inline tests/simple/testdata/integer_math.textproto]: s/devide/divide/

File: tests/simple/testdata/integer_math.textproto (hunk 2 above)
Comment (madi8229) [reply tests/simple/testdata/integer_math.textproto]: done

**Comment by madi8229** [comment]
PTAL.

**Pull request merged**